sw/device/lib/base/log.h

```diff
@@ -100,52 +91,58 @@ typedef struct log_fields {
-          .file_name = kFileNameStr,                                          \
-          .line = __LINE__,                                                   \
-          .nargs = GET_NUM_VARIABLE_ARGS(_format, ##__VA_ARGS__),             \
-          .format = "" _format ""};                                           \
-      base_log_internal_dv(&kLogFields,                                       \
-                           GET_NUM_VARIABLE_ARGS(_format, ##__VA_ARGS__),     \
-                           ##__VA_ARGS__);                                    \
-    } else {                                                                  \
-      base_log_internal_core(_severity, __FILE__, __LINE__, "" _format "",    \
-                             _format, ##__VA_ARGS__);                         \
-    }                                                                         \
+#define LOG(severity, format, ...) LOG_INNER_(severity, format, ##__VA_ARGS__)
+
+/**
+ * Implementation detail of `LOG`.
+ */
+#define LOG_INNER_(_severity, _format, ...)                                 \
+  do {                                                                      \
+    if (kDeviceType == kDeviceSimDV) {                                      \
+      /* clang-format off */                                                \
+      /* Put DV-only log constants in .logs.* sections, which the
+       * linker will dutifully discard.
+       * Unfortunately, clang-format really mangles these declarations,
+       * so we format them manually. */                                     \
+      __attribute__((section(".logs.strings")))                             \
+      static const char kFileNameStr[] = "" __FILE__ "";                    \
+      __attribute__((section(".logs.fields")))                              \
+      static const log_fields_t kLogFields = {                              \
+          .severity = _severity,                                            \
+          .file_name = kFileNameStr,                                        \
+          .line = __LINE__,                                                 \
+          .nargs = GET_NUM_VARIABLE_ARGS(__VA_ARGS__),                      \
+          .format = "" _format "",                                          \
+      }; /* clang-format on */                                              \
+                                                                            \
+      base_log_internal_dv(&kLogFields, GET_NUM_VARIABLE_ARGS(__VA_ARGS__), \
+                           ##__VA_ARGS__);                                  \
+    } else {                                                                \
+      log_fields_t log_fields = {                                           \
+          .severity = _severity,                                            \
+          .file_name = "" __FILE__ "",                                      \
```

Comment:
This will put the file name in `.rodata` anyway. Is there even a need for the DV flow to create a separate `.logs.strings` section?

Reply:
Yeah, that's a good point. Should I just get rid of it?

Reply:
Yes, I am fine with removing it. Remove it from the python script invocation as well here:
`hw/dv/data/sim.mk` lines 55 and 63.

Reply:
Done.